### PR TITLE
267 unsynced records retry and table

### PR DESCRIPTION
**Pull request by HaleyBoles**
Added Unsynced Records Route/Page
- Only shows in sidebar if there are unsynced records
- shows count of unsynced records in sidebar
- grid of unsynced records. 
   - click to show record details in grid
   - click to show error message in grid
   - button to clear single error in grid
        - errors automatically if the error code is Washington or Roane since these have to be finger-fixed
        - later on when we add debug mode it can skip error so can be cleared after fixing. For now, have to remove error in DB or use "clear all" 
   - button to clear all errors
   - button to sync
   - lots of TODOs for future additions like single record sync

Linked unsynced icon in Ticket History to Unsynced Records page 
Page header className cleanup
Button disabled hover icon only cleanup

### CODE REVIEW

File: src/lib/graphql/mutations.ts [hunk 1]
```diff
@@ -397,7 +397,7 @@ export const START_LISTENING_LPR_CAMERA = graphql(/* GraphQL */ `
 `);
 
 export const STOP_LISTENING_LPR_CAMERA = graphql(/* GraphQL */ `
-	mutation SSTOP_LISTENING_LPR_CAMERA($input: StopListeningLprCameraInput!) {
+	mutation STOP_LISTENING_LPR_CAMERA($input: StopListeningLprCameraInput!) {
```
Comment (sspoon865) [inline src/lib/graphql/mutations.ts:400]: Nice catch. 😎

File: src/routes/UnsyncedRecordsRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,854 @@
...
+		date: Date;
+		details: Object;
+		syncedError: string;
+		id: number;
+	}
+
+	const unsyncedRecordsData: UnsyncedRecords[] = useMemo(() => {
+		let unsyncedRecords: UnsyncedRecords[] = [];
+		// Map unsynced tickets
+		unsyncedTickets?.map((ticket) => {
```
Comment (sspoon865) [inline src/routes/UnsyncedRecordsRoute.tsx:99]: Minor nitpick
You should already receive back a result of the callback function for each element in the array when using `.map()`, it is probably not necessary to use `Object.assign` and then `push()`

This section could be simplified to be clearer later on.

File: src/routes/UnsyncedRecordsRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,854 @@
...
+	}, [unsyncedTicketsQuery, unsyncedTrucksQuery, unsyncedTrucksInYardQuery, unsyncedTruckInYardHistoryQuery]);
+
+	const numberOfErrors = unsyncedRecordsData
+		.map((r) => {
+			if (r.syncedError != "" && r.syncedError != null) {
+				return r.syncedError;
+			}
+			return;
+		})
+		.filter((r) => r).length;
```
Comment (sspoon865) [inline src/routes/UnsyncedRecordsRoute.tsx:340]: `.map()` is probably unnecessary here, you can achieve correct length with chained `.filter()` function
```suggestion
		.filter((r) => r.syncedError)).length
```

File: src/routes/UnsyncedRecordsRoute.tsx [hunk 4]
```diff
@@ -0,0 +1,854 @@
...
+			{
+				field: "details",
+				headerName: "Record ID",
+				minWidth: 120,
+				valueFormatter: (params) => {
+					const data = Object.entries(params.value);
+					const record = data
+						.find((d) => {
+							const dLower = d[0].toLowerCase();
+							return dLower == "ticket #" || dLower == "truck id";
```
Comment (sspoon865) [inline src/routes/UnsyncedRecordsRoute.tsx:370]: Should use === in almost all cases
```suggestion
							return dLower === "ticket #" || dLower === "truck id";
```

**Review by sspoon865 — CHANGES_REQUESTED** [review]
Just a couple smol changes.

I am assuming you were able to test a lot of it as well? I didn't try to create any sync issues, but the code looks good.

File: src/routes/UnsyncedRecordsRoute.tsx (hunk 2 above)
Comment (HaleyBoles) [reply src/routes/UnsyncedRecordsRoute.tsx:99]: @sspoon865 Hmm this is the only way I could get it to work to assign a new object line to the object within an object: 
{
   type: "ticket",
   details: {
      1: val,
      2: val,
      [assigning more here]
   }
    id: id
}

Then once all that extra stuff is conditionally added then I'm shoving it into the array.

May need to have you look at with me lol

Comment (HaleyBoles) [reply src/routes/UnsyncedRecordsRoute.tsx:99]: skipping for now, may look at condensing later

**Review by sspoon865 — APPROVED** [review]

**Pull request merged**